aws/dx_vif.go

```diff
@@ -0,0 +1,256 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/arn"
+	"github.com/aws/aws-sdk-go/service/directconnect"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+)
+
+func isNoSuchDxVirtualInterfaceErr(err error) bool {
+	return isAWSErr(err, "DirectConnectClientException", "does not exist")
+}
+
+func dxVirtualInterfaceRead(d *schema.ResourceData, meta interface{}) (*directconnect.VirtualInterface, error) {
```

Comment:
Do you mind reducing the interface to bare minimum here? e.g. `(id string, conn *directconnect.DirectConnect)`?

Exceptions apply, but ideally only CRUD should be responsible for modifying the state, i.e. calling `d.SetId("")`.

Reply:
Will do.

Reply:
Done. I also changed the checking to just remove the resource from state when the status is `Deleted` and not `Deleting` or `Rejected`, similar to https://github.com/terraform-providers/terraform-provider-aws/pull/2990.